### PR TITLE
Prevent panic if fileinfo is nil

### LINKED ISSUE #4845: filecount plugin: invalid memory address or nil pointer dereference
Hi,

Telegraph app throw excetion about filecount plugin after then service is stop.

## System info

Windows Server 2012 R2 v6.3 build 3600

## Config

```
[[inputs.filecount]]
directory = "C:\\app\\data\\que"
name = "*.log"
recursive = false
regular_only = false
size = 0
```

## Error
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x18 pc=0xce7d41]

goroutine 216 [running]:
github.com/influxdata/telegraf/plugins/inputs/filecount.count.func1(0xc04288e380, 0x3a, 0x0, 0x0, 0x2190fa0, 0xc042716090, 0x0, 0x0)
        /go/src/github.com/influxdata/telegraf/plugins/inputs/filecount/filecount.go:134 +0x81
path/filepath.walk(0xc042386620, 0x15, 0x21afd20, 0xc042a70000, 0xc042716030, 0x0, 0x30)
        /usr/local/go/src/path/filepath/path.go:377 +0x214
path/filepath.Walk(0xc042386620, 0x15, 0xc042716030, 0x1, 0xc042716000)
        /usr/local/go/src/path/filepath/path.go:403 +0x10d
github.com/influxdata/telegraf/plugins/inputs/filecount.count(0xc042386620, 0x15, 0x0, 0xc042716000, 0x0, 0x0)
        /go/src/github.com/influxdata/telegraf/plugins/inputs/filecount/filecount.go:139 +0x8f
github.com/influxdata/telegraf/plugins/inputs/filecount.(*FileCount).Gather(0xc0420c8320, 0x21b0c80, 0xc042122680, 0x0, 0xc04237e000)
        /go/src/github.com/influxdata/telegraf/plugins/inputs/filecount/filecount.go:183 +0xdc
github.com/influxdata/telegraf/agent.gatherWithTimeout.func1(0xc04208a360, 0xc042094e40, 0x21b0c80, 0xc042122680)
        /go/src/github.com/influxdata/telegraf/agent/agent.go:153 +0x4e
created by github.com/influxdata/telegraf/agent.gatherWithTimeout
        /go/src/github.com/influxdata/telegraf/agent/agent.go:152 +0xdb
```
## Expected behavior:

telegraf windows service must  be work ^_^

## Actual behavior:

Service is stop

### PR DESCRIPTION
**Pull request by glinton**
Resolves #4845

### CODE REVIEW

File: plugins/inputs/filecount/filecount.go [hunk 1]
```diff
@@ -127,6 +128,9 @@ func absDuration(x time.Duration) time.Duration {
 
 func count(basedir string, recursive bool, countFn countFunc) error {
 	walkFn := func(path string, file os.FileInfo, err error) error {
+		if file == nil {
+			return fmt.Errorf("Nil file found at %q", path)
```
Comment (danielnelson) [inline plugins/inputs/filecount/filecount.go]: Check the err argument and return it, I think it should have a better error string.

**Comment by c1982** [comment]
is this patch active in nightly build?

**Comment by danielnelson** [comment]
@c1982 Not just yet, but I can give you a link to a CI build if you tell me which one you need.

**Comment by c1982** [comment]
amd64/Windows

**Comment by danielnelson** [comment]
I imagine you will still have an error reported to the log, just no more crash.  It would be interesting to see what is logged.  Here is the build:
- [telegraf-1.9.0%7E8f67a2ec_windows_amd64.zip](https://20874-33258973-gh.circle-artifacts.com/0/build/windows/amd64/telegraf-1.9.0%7E8f67a2ec_windows_amd64.zip)

**Comment by c1982** [comment]
Yes. I tested your version in production. Local service it was not crash. is good. 
Also, like you said. Still have an error in log file. 

May be help for fix:

`2018-10-12T02:22:20Z E! Error in plugin [inputs.filecount]: GetFileAttributesEx Z:\Que\d99be30c85ba478b8a50fe57e52dd78b.eml: The system cannot find the file specified.`

This is SMTP queue. So, file is very quickly created and deleted in folder. Plugin try read file attributes but file is gone. 

Is possible will add config parameters for attributes reading disabled?

**Pull request merged**